include/onnxruntime/core/platform/EigenNonBlockingThreadPool.h

```diff
@@ -767,6 +767,29 @@ class ThreadPoolTempl : public onnxruntime::concurrency::ExtendedThreadPoolInter
   typedef std::function<void()> Task;
   typedef RunQueue<Task, Tag, 1024> Queue;
 
+  // Class for waiting w/ exponential backoff.
+  // Template argument is maximum number of spins in backoff loop.
+  template <unsigned kMaxBackoff>
+  class ThreadPoolWaiter {
+    // Current number if spins in backoff loop
+    unsigned pause_time_;
```

Comment:
Where do we initialize the pause_time_? 
Normally we have constructor, but this class has not defined constructor.

Reply:
Its default constructed at every point it is created no?

Reply:
No.  It wouldn't .
If you write:
```
std::string s;
```
`s` will be default constructed, as std::string a type with a default constructor. However, when you write:
```
int s;
```
`s` will be default constructed. Because `int` is not a class. It is a builtin type. It doesn't have a  default constructor. All numeric types are like this.

Reply:
I mean wherever we use `ThreadPoolWaiter` we do: `ThreadPoolWaiter<...> waiter{}` the `{}` is what default initializes.

Reply:
For class objects: Using {} will invoke the default constructor if one is provided. If no constructor is defined, the compiler will generate one (which is often a no-op), and the member variables will be initialized as follows:
* If the members are primitive types (e.g., int, float) and are not explicitly initialized in the constructor or the member initializer list, they will not be initialized and will contain garbage values.

Reply:
Is this not [value initializing](https://en.cppreference.com/w/cpp/language/value_initialization) in which case members are value initialized (which for `int` is zerod).

Either way its easy enough to make explicit.

Reply:
Right. Based on the spec of value -initialization, the int will have zero value.